ocp_resources/resource.py

```diff
@@ -134,9 +134,47 @@ def sub_resource_level(current_class: Any, owner_class: Any, parent_class: Any)
     return None
 
 
-# Exceptions classes
+def replace_key_with_hashed_value(resource_dict: Dict[Any, Any], key_name: str) -> Dict[Any, Any]:
+    """
+    Recursively search a nested dictionary for a given key and changes its value to "******" if found.
+    The function supports two key formats:
+    1. Regular path: "path>to>key"
+    2. List path: "path>to[]>key"
+    Args:
+        resource_dict: The nested dictionary to search. Must be a dictionary.
```

Comment:
```suggestion
        resource_dict: The nested dictionary to search. 
```
no need for this; the function has typing